### ISSUE TITLE
sentinel support broken?

### ISSUE THREAD

**Issue by KlavsKlavsen** [arioch/puppet-redis#166]
When doing this:
    class { 'redis::sentinel':
      master_name      => $master_name,
      redis_host       => $redis_master,
      failover_timeout => $failover_timeout,
    }

  class { 'redis':
    bind        => $listen_ip,
    slaveof     => $redis_master ? {
        "$fqdn" => undef,
        default => "$redis_master 6379"
    }
  }

to setup my hosts - with one being set to master-  and redis-sentinel setup on same host.. puppet dies with:
 Duplicate declaration: Package[redis-server] - as its both defined in install.pp and in redis::sentinel

I assume someone have used redis with sentinel with this module before, and I'm just understanding the documentation wrong?

**Comment by KlavsKlavsen** [comment]
This is puppet 4 master and my servers are ubuntu 16.04

**Comment by petems** [comment]
@KlavsKlavsen I haven't looked at the sentinal work for a while, could you try the following change:

```diff
diff --git a/manifests/sentinel.pp b/manifests/sentinel.pp
index 617dfc4..298c647 100644
--- a/manifests/sentinel.pp
+++ b/manifests/sentinel.pp
@@ -190,11 +190,7 @@ class redis::sentinel (
   $client_reconfig_script = $::redis::params::sentinel_client_reconfig_script,
 ) inherits redis::params {
 
-  unless defined(Package[$package_name]) {
-    ensure_resource('package', $package_name, {
-      'ensure' => $package_ensure
-    })
-  }
+  require redis
```

**Comment by petems** [comment]
I'm also adding in an acceptance test for a basic sentinel setup 👍

**Comment by KlavsKlavsen** [comment]
that gives  Class[Redis] is already declared; Duplicate declaration: Class[Redis] is already declared; cannot redeclare.. if I remove it - and don't insert "require redis" - then it runs.

Next is that redis::sentinel is adding an init script (kinda odd when in systemd land :) - and when systemd starts it - it says all is good.. except /usr/sbin/redis-sentinel binary does not exist.. I'm assuming the sentinel support actually works that way as sentinel was seperate from redis before 3.0? I assume module was suppose to create symlink /usr/bin/redis-sentinel - pointing to redis binary? (or just run redis with --sentinel option) ?

**Comment by KlavsKlavsen** [comment]
Are you running sentinel cluster mode instead?

**Comment by petems** [comment]
Basically, this is because of different packaging standards:

```
[root@master redskull-controller]# yum provides /bin/redis-sentinel
Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * base: mirror.cov.ukservers.com
 * epel: mirror.logol.ru
 * extras: mirror.sax.uk.as61049.net
 * updates: mirrors.vooservers.com
redis-3.2.3-1.el7.x86_64 : A persistent key-value database
Repo        : @epel
Matched from:
Filename    : /bin/redis-sentinel
```

So you could use Puppet to create this symlink.

I would say I don't know enough about sentinel to accurately test this. I will open a PR to change the package requirement to be an include of redis.

**Comment by petems** [comment]
I believe this was fixed by #171 and an additional acceptance test was added, if not I'll re-open
